### ISSUE TITLE
Does Sketchup.get_shortcuts return values in localized text ?

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#529]
### SketchUp Ruby API Question

**Does [`Sketchup.get_shortcuts`](https://ruby.sketchup.com/Sketchup.html#get_shortcuts-class_method) return values in localized text ?**

The documentation does not say. I'm needing both the keychords and the command text localized s the user will use and know them.

BTW the **backslash** for the tab character is missing from the example.

**Comment by DanRathbun** [comment]
Apparently not. It looks like it's half and half.
The keys in localized lang, but the menu paths in English.

Also to complicate things, on the Mac the command strings are mostly all send actions.

**Comment by thomthom** [comment]
Do you want to make this into a feature request?

**Comment by DanRathbun** [comment]
**Probably.** But there are also bug issues with certain English shortcuts "leaking" through. Andreas thinks these are the hardcoded default shortcuts, yet not all of them "leak" through. So I need to compile a list of the "leakers".

The feature would be that on both platforms, localized menu paths (and keychords) are returned by the method.
In the meantime, for Mac, I am creating a lookup hash from John Boundy's Mac send action list he posted over at SketchUcation years ago. (I had to switch to a simpler web dialog example so I'm off of the Shortcut Lister example for awhile.)

**Comment by thomthom** [comment]
The `get_shortcuts` method, as it is now, isn't very API friendly. But even the Export/Import shortcut feature in SU isn't very robust between platforms or even SketchUp versions. I'd ideally like to see these exposed with more platform, version and language agnostic parameters.

**Comment by DanRathbun** [comment]
> The [`get_shortcuts`](https://ruby.sketchup.com/Sketchup.html#get_shortcuts-class_method) method, as it is now, isn't very API friendly. 

Perhaps not, but coders have found it useful to "discover" unpublished [send_action](https://ruby.sketchup.com/Sketchup.html#send_action-class_method) strings that can be used for toolbar buttons, context menus, (usually on Mac platform.)

> _**Aside:** It would also be nice if there was a formal API means to discover [send_action](https://ruby.sketchup.com/Sketchup.html#send_action-class_method) strings._

I think if the **current method** were changed, that it might break "workflows" at the least.

**The current method can (and probably should) remain as is for backward compatibility.**

And I don't know if it's worth it to open another issue for bugs I find for the existing method (although the method doc doesn't guarantee any translation much less proper localization.) The fact that it returns send action strings on Mac is seen as a benefit to some.

----
So we'd need a **new** sister method ...

> **`Sketchup.get_localized_shortcuts`**
> Returns (correct for the running platform) all keychord and command menupaths in proper localized text for display in extension interface(s).
> (_Possibly a boolean override argument to return in the API language which is English._)

I'll retitle this as a Feature Request for a new better API method.

**Comment by thomthom** [comment]
I'd really really like to see any usage of `send_action` obsolete with proper APIs. So please file feature requests for `send_action` usage that is being relied on.

**Comment by DanRathbun** [comment]
> So please file feature requests for send_action usage that is being relied on.

Right, but **this** request results **because** send actions are "relied on" (for return from an API call.)
